### PR TITLE
Describe component metadata templating possibilities

### PR DESCRIPTION
**Pull request by VerstraeteBert**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

## Description

Describe templating possibilities for metadata values. [Before, only {uuid} and {podName} were described](https://docs.dapr.io/operations/components/component-schema/#special-metadata-values). Now, {namespace} and {appID} are noted as additional possibilities.

## Issue reference

Linked to https://github.com/dapr/dapr/pull/5752

### CODE REVIEW

File: daprdocs/content/en/operations/components/component-schema.md [hunk 1]
```diff
@@ -44,7 +44,16 @@ spec:
 
 ### Special metadata values
```
Comment (msfussell) [inline daprdocs/content/en/operations/components/component-schema.md]:
```suggestion
### Templated metadata values
```

File: daprdocs/content/en/operations/components/component-schema.md [hunk 2]
```diff
@@ -44,7 +44,16 @@ spec:
...
 
-Metadata values can contain a `{uuid}` tag that is replaced with a randomly generate UUID when the Dapr sidecar starts up. A new UUID is generated on every start up. It can be used, for example, to have a pod on Kubernetes with multiple application instances consuming a [shared MQTT subscription]({{< ref "setup-mqtt.md" >}}). Below is an example of using the `{uuid}` tag.
+Metadata values can contain template tags that are resolved on Dapr sidecar startup. The table below shows the current possibilities for templating Components.
```
Comment (msfussell) [inline daprdocs/content/en/operations/components/component-schema.md]:
```suggestion
Metadata values can contain template tags that are resolved on Dapr sidecar startup. The table below shows the current templating tags that can be used in components.
```

File: daprdocs/content/en/operations/components/component-schema.md [hunk 3]
```diff
@@ -44,7 +44,16 @@ spec:
...
+|-------------|------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| {uuid}      | Randomly generated UUIDv4                                        | Need for a unique identifier in self-hosted mode, for example, multiple application instances consuming a [shared MQTT subscription]({{< ref "setup-mqtt.md" >}}) |
+| {podName}   | Name of the Pod containing the Dapr sidecar                      | This can be used to have a persisted behavior where the ConsumerID does not change on restart when using StatefulSets in Kubernetes                               |
+| {namespace} | Namespace where the Dapr sidecar resides combined with its appId | Using a shared clientId when multiple application instances consume a Kafka topic in Kubernetes                                                                   |
```
Comment (msfussell) [inline daprdocs/content/en/operations/components/component-schema.md]:
```suggestion
| {namespace} | Namespace where the Dapr sidecar resides combined with its appId | Using a shared `clientId` when multiple application instances consume a Kafka topic in Kubernetes                                                                   |
```

File: daprdocs/content/en/operations/components/component-schema.md [hunk 4]
```diff
@@ -44,7 +44,16 @@ spec:
...
+| {appID}     | The configured appID of the resource containing the Dapr sidecar | Having a shared clientId when multiple application instances consumer a Kafka topic in self-hosted mode                                                           |
```
Comment (msfussell) [inline daprdocs/content/en/operations/components/component-schema.md]:
```suggestion
| {appID}     | The configured `appID` of the resource containing the Dapr sidecar | Having a shared `clientId` when multiple application instances consumer a Kafka topic in self-hosted mode                                                           |
```

**Review by msfussell — COMMENTED** [review]
@VerstraeteBert - Great contributions. I suggest that you also cross-reference the Kafka examples from the Kafka component documentation with an example in the Kafka component docs. That way people are able to understand how to use this templated value. Would you be able to add this? (a link to this article and an example)?

File: daprdocs/content/en/operations/components/component-schema.md [hunk 5]
```diff
@@ -44,7 +44,16 @@ spec:
...
+Metadata values can contain template tags that are resolved on Dapr sidecar startup. The table below shows the current possibilities for templating Components.
+
+| Tag         | Details                                                          | Example use case                                                                                                                                                  |
+|-------------|------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| {uuid}      | Randomly generated UUIDv4                                        | Need for a unique identifier in self-hosted mode, for example, multiple application instances consuming a [shared MQTT subscription]({{< ref "setup-mqtt.md" >}}) |
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-schema.md]:
```suggestion
| {uuid}      | Randomly generated UUIDv4                                        | When you need a unique identifier in self-hosted mode; for example, multiple application instances consuming a [shared MQTT subscription]({{< ref "setup-mqtt.md" >}}) |
```

File: daprdocs/content/en/operations/components/component-schema.md [hunk 6]
```diff
@@ -44,7 +44,16 @@ spec:
...
+| {podName}   | Name of the Pod containing the Dapr sidecar                      | This can be used to have a persisted behavior where the ConsumerID does not change on restart when using StatefulSets in Kubernetes                               |
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-schema.md]:
```suggestion
| {podName}   | Name of the pod containing the Dapr sidecar                      | Use to have a persisted behavior, where the ConsumerID does not change on restart when using StatefulSets in Kubernetes                               |
```

**Comment by VerstraeteBert** [comment]
@msfussell @hhunter-ms Thanks for the feedback, a lot more pleasant to read now :-). 

I've added my suggestion for how it can look on the kafka pubsub page.

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md [hunk 7]
```diff
@@ -11,6 +11,9 @@ aliases:
 
 To setup Apache Kafka pubsub create a component of type `pubsub.kafka`. See [this guide]({{< ref "howto-publish-subscribe.md#step-1-setup-the-pubsub-component" >}}) on how to create and apply a pubsub configuration. For details on using `secretKeyRef`, see the guide on [how to reference secrets in components]({{< ref component-secrets.md >}}).
 
+All component metadata field values can carry [special metadata values]({{< ref "component-schema.md#templated-metadata-values" >}}) carry special metadata values, which are resolved on dapr sidecar startup.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md]:
```suggestion
Component metadata field values can contain [templated metadata values]({{< ref "component-schema.md#templated-metadata-values" >}}) which are resolved on Dapr sidecar startup. 
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md [hunk 8]
```diff
@@ -11,6 +11,9 @@ aliases:
 
 To setup Apache Kafka pubsub create a component of type `pubsub.kafka`. See [this guide]({{< ref "howto-publish-subscribe.md#step-1-setup-the-pubsub-component" >}}) on how to create and apply a pubsub configuration. For details on using `secretKeyRef`, see the guide on [how to reference secrets in components]({{< ref component-secrets.md >}}).
 
+All component metadata field values can carry [special metadata values]({{< ref "component-schema.md#templated-metadata-values" >}}) carry special metadata values, which are resolved on dapr sidecar startup. 
+For example, you can choose to use `{namespace}` as the `consumerGroup`, providing a unique consumer group linked to both the Kubernetes `namespace` the sidecar is in and the configured `appId`.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md]:
```suggestion
For example, you can choose to use `{namespace}` as the `consumerGroup`, providing a unique consumer group associated to both the Kubernetes `namespace` the sidecar is in and the configured `appId`.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md [hunk 9]
```diff
@@ -23,7 +26,7 @@ spec:
   - name: brokers # Required. Kafka broker connection setting
     value: "dapr-kafka.myapp.svc.cluster.local:9092"
   - name: consumerGroup # Optional. Used for input bindings.
-    value: "group1"
+    value: "{namespace}"
   - name: clientID # Optional. Used as client tracing ID by Kafka brokers.
     value: "my-dapr-app-id"
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md:31]: Question: @VerstraeteBert - Could you use `{appID} `here as a templated value, and if so why/when. Does this have value?

Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md:31]: @VerstraeteBert - Added a couple more comments and a question above^^

Comment (VerstraeteBert) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md:31]: {appId} makes sense to use as as a unique, human-readable identifier of an application when there is no concept of namespaces, like in self-host mode. This [quality article](https://docs.dapr.io/developing-applications/building-blocks/pubsub/howto-namespace/#with-namespace-consumer-groups) implies that {appId} is used as the default for consumerGroup. I'll link to that article in the doc.

Comment (VerstraeteBert) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md:31]: Ouch, been forgetting to sign-off... Recreating the PR will be easier than going through the rebase at this point (multiple collaborators). My apologies.

**Pull request closed without merging**